benchmark-ph1ex3/neutronics/convert.py

```diff
@@ -0,0 +1,2493 @@
+            for gp in range(G):
+                ss = 0
+                if g == 0 and gp == 0:
+                    for i in range(lim[0]):
+                        for j in range(lim[0]):
+                            ss += float(HXS['SP0'][i, j]) * \
+                                float(HXS['FLX'][i])
+
+                elif g == 0:
+                    for i in range(lim[0]):
+                        for j in range(lim[gp-1], lim[gp]):
+                            ss += float(HXS['SP0'][i, j]) * \
+                                float(HXS['FLX'][i])
+
+                elif gp == 0:
+                    for i in range(lim[g-1], lim[g]):
+                        for j in range(lim[0]):
+                            ss += float(HXS['SP0'][i, j]) * \
+                                float(HXS['FLX'][i])
+
+                else:
+                    for i in range(lim[g-1], lim[g]):
+                        for j in range(lim[gp-1], lim[gp]):
+                            ss += float(HXS['SP0'][i, j]) * \
+                                float(HXS['FLX'][i])
+
+                CXS['SP0'][g, gp] = ss/CXS['FLX'][g]
+
+        # Use this to homogenize the diff coeficient
+        data = ['FLX', 'ST', 'DIFFCOEF']
+        for param in data:
+            f = open(base + param + '.txt', 'a')
+            f.write(str(float(temp)-273))
+            for dg in CXS[param]:
+                f.write(' ' + str(dg))
+            f.write('\n')
+            f.close()
+
+        # # Uncomment this to homogenize TR cross section
+        # f = open(base + 'DIFFCOEF' + '.txt', 'a')
+        # f.write(str(temp))
+        # for dg in HXS['TR']:
+        #     f.write(' ' + str(1./3./dg))
+        # f.write('\n')
+        # f.close()
+
+        f = open(base + 'SP0' + '.txt', "w+")
+        f.write(str(float(temp)-273))
+        for i in range(len(CXS['SP0'])):
+            for dg in CXS['SP0'][i]:
+                f.write(' ' + str(dg))
+        f.write('\n')
+        f.close()
+        # print('SP0 done')
+
+        f = open(base + 'REMXS' + '.txt', 'a')
+        f.write(str(float(temp)-273))
+        for i in range(len(CXS['SP0'])):
+            scatii = float(CXS['SP0'][i, i])
+            toti = float(CXS['ST'][i])
+            f.write(' ' + str(toti-scatii))
+        f.write('\n')
+        f.close()
+
+        # Everythin that is zero
+        data = ['NSF', 'FISS', 'CHIT', 'KAPPA', 'INVV', 'CHID']
+        for param in data:
+            f = open(base + param + '.txt', 'a')
+            f.write(str(float(temp)-273))
+            for dg in range(G):
+                f.write(' ' + str(0))
+            f.write('\n')
+            f.close()
+
+        data = ['BETA_EFF', 'LAMBDA']
+        for param in data:
+            f = open(base + param + '.txt', 'a')
+            f.write(str(float(temp)-273))
+            for i in range(8):
+                f.write(' 0')
+            f.write('\n')
+            f.close()
+
+
+def homoge_collapse_refl(directory):
```

Comment:
Similar comment as above. You have some functions that create group constants for different components (reflector, fuel) and functions that homogenize components (reflector, fuel).

This breaks the "Don't Repeat Yourself" (D.R.Y.) best practice. Try making one function that can be adjusted to do different tasks.